Lib/asyncio/base_events.py

```diff
@@ -463,21 +463,25 @@ def create_task(self, coro, *, name=None, context=None):
 
         Return a task object.
         """
-        self._check_closed()
-        if self._task_factory is None:
-            task = tasks.Task(coro, loop=self, name=name, context=context)
-            if task._source_traceback:
-                del task._source_traceback[-1]
-        else:
-            if context is None:
-                # Use legacy API if context is not needed
-                task = self._task_factory(self, coro)
+        task = None
+        try:
+            self._check_closed()
+            if self._task_factory is None:
+                task = tasks.Task(coro, loop=self, name=name, context=context)
+                if task._source_traceback:
+                    del task._source_traceback[-1]
             else:
-                task = self._task_factory(self, coro, context=context)
+                if context is None:
+                    # Use legacy API if context is not needed
+                    task = self._task_factory(self, coro)
+                else:
+                    task = self._task_factory(self, coro, context=context)
```

Comment:
The sentinel for `context` is `None` already, right? I think we can simplify this down to:
```suggestion
                task = self._task_factory(self, coro, context=context)
```

Reply:
No this doesn't work if the factory does not accept the context kwarg